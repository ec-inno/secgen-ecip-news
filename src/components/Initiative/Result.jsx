import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import axios from 'axios';
import { Link } from 'gatsby';
import { useTranslation } from 'react-i18next';

import Icon from '../Icon';
import defaultImage from '../assets/images/default-image.png';
import formatStatus from '../../utils/formatStatus';

const Result = ({ title, href, status, pubRegNum, logo }) => {
  const { t } = useTranslation();
  const [logoData, setLogoData] = useState('');
  const { GATSBY_INITIATIVES_API: api } = process.env;

  const hasLogo = logo && logo.id && logo.mimeType;

  useEffect(() => {
    if (hasLogo) {
      axios
        .get(`${api}/register/logo/${logo.id}`, {
          responseType: 'arraybuffer',
        })
        .then(r => {
          const data = Buffer.from(r.data, 'binary').toString('base64');
          setLogoData(data);
        })
        .catch(e => {
          console.error(`Failed to fetch logo ${logo.id}`, e);
        });
    }
  }, []);

  const background = hasLogo
    ? `data:${logo.mimeType};base64,${logoData}`
    : defaultImage;

  return (
    <article className="ecl-u-d-flex ecl-u-border-bottom ecl-u-border-color-grey-15 ecl-u-pv-m">
      <div className="ecl-u-flex-grow-1">
        <div className="ecl-u-type-prolonged-m ecl-u-type-bold ecl-u-mt-xs">
          <Link
            to={href}
            className="ecl-u-type-prolonged-m ecl-u-type-bold ecl-link"
          >
            {title}
          </Link>
        </div>
        <ul className="ecl-u-mt-m ecl-unordered-list ecl-unordered-list--no-bullet">
          <li className="ecl-u-d-flex ecl-u-align-items-center ecl-u-type-color-grey-75 ecl-unordered-list__item">
            <Icon className="ecl-icon--m" shape="general--organigram" />
            <span className="ecl-u-type-s ecl-u-ml-s">
              {t('Current status')}
              {': '}
              {formatStatus(status)}
            </span>
          </li>
          <li className="ecl-u-d-flex ecl-u-align-items-center ecl-u-type-color-grey-75 ecl-unordered-list__item">
            <Icon className="ecl-icon--m" shape="general--edit" />
            <span className="ecl-u-type-s ecl-u-ml-s">
              {t('Registration number')}
              {': '}
              {pubRegNum}
            </span>
          </li>
        </ul>
      </div>
      <div
        role="img"
        aria-label={title}
        className="ecl-u-media-ratio-3-2 ecl-u-flex-shrink-0 ecl-u-ml-xl ecl-u-d-lg-none"
        style={{
          backgroundImage: `url('${background}')`,
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'contain',
          height: '5rem',
          width: '7.5rem',
        }}
      ></div>
      <div
        role="img"
        aria-label={title}
        className="ecl-u-media-ratio-3-2 ecl-u-flex-shrink-0 ecl-u-ml-xl ecl-u-d-none ecl-u-d-lg-block"
        style={{
          backgroundImage: `url('${background}')`,
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'contain',
          height: '8.75rem',
          width: '13.125rem',
        }}
      ></div>
    </article>
  );
};

Result.propTypes = {
  title: PropTypes.string,
  href: PropTypes.string,
  status: PropTypes.string,
  pubRegNum: PropTypes.string,
  logo: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    mimeType: PropTypes.string,
    size: PropTypes.number,
  }),
};

export default Result;
