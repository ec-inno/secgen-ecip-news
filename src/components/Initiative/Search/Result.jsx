import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

const Result = ({ title, href, pubRegNum }) => {
  if (!title) return '';

  const { t } = useTranslation();

  return (
    <article className="ecl-u-type-m ecl-u-mt-l ecl-u-pb-l ecl-u-pb-lg-m">
      <a
        href={href}
        className="ecl-u-type-prolonged-m ecl-u-type-bold ecl-link"
      >
        {title}
      </a>
      <p className="ecl-u-type-paragraph-m ecl-u-type-color-grey ecl-u-mv-none">
        {t('Registration number')}
        {': '}
        {pubRegNum}
      </p>
    </article>
  );
};

Result.propTypes = {
  title: PropTypes.string,
  href: PropTypes.string,
  pubRegNum: PropTypes.string,
};

export default Result;
