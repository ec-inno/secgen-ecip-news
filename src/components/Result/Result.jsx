/**
 * Based on "Content item" composition
 * @see https://ec.europa.eu/component-library/ec/templates/compositions/content-item/usage/
 */
import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'gatsby';
import { useTranslation } from 'react-i18next';

import upperCaseFirstChar from '@eci/utils/upperCaseFirstChar';

import Icon from '../Icon';

const Result = ({ title, background, href, status, pubRegNum }) => {
  const { t } = useTranslation();

  return (
    <article className="ecl-u-d-flex ecl-u-border-bottom ecl-u-border-color-grey-15 ecl-u-pv-m">
      <div className="ecl-u-flex-grow-1">
        <div className="ecl-u-type-prolonged-m ecl-u-type-bold ecl-u-mt-xs">
          <Link
            to={href}
            className="ecl-u-type-prolonged-m ecl-u-type-bold ecl-link"
          >
            {title}
          </Link>
        </div>
        <ul className="ecl-u-mt-m ecl-unordered-list ecl-unordered-list--no-bullet">
          <li className="ecl-u-d-flex ecl-u-align-items-center ecl-u-type-color-grey-75 ecl-unordered-list__item">
            <Icon className="ecl-icon--m" shape="general--organigram" />
            <span className="ecl-u-type-s ecl-u-ml-s">
              {t('Current status')}
              {': '}
              {upperCaseFirstChar(status)}
            </span>
          </li>
          <li className="ecl-u-d-flex ecl-u-align-items-center ecl-u-type-color-grey-75 ecl-unordered-list__item">
            <Icon className="ecl-icon--m" shape="general--edit" />
            <span className="ecl-u-type-s ecl-u-ml-s">
              {t('Registration number')}
              {': '}
              {pubRegNum}
            </span>
          </li>
        </ul>
      </div>
      <div
        role="img"
        aria-label={title}
        className="ecl-u-media-ratio-3-2 ecl-u-flex-shrink-0 ecl-u-ml-xl ecl-u-d-lg-none"
        style={{
          backgroundImage: `url('${background}')`,
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'contain',
          height: '5rem',
          width: '7.5rem',
        }}
      ></div>
      <div
        role="img"
        aria-label={title}
        className="ecl-u-media-ratio-3-2 ecl-u-flex-shrink-0 ecl-u-ml-xl ecl-u-d-none ecl-u-d-lg-block"
        style={{
          backgroundImage: `url('${background}')`,
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'contain',
          height: '8.75rem',
          width: '13.125rem',
        }}
      ></div>
    </article>
  );
};

Result.propTypes = {
  title: PropTypes.string,
  href: PropTypes.string,
  status: PropTypes.string,
  pubRegNum: PropTypes.string,
  background: PropTypes.string,
};

export default Result;
