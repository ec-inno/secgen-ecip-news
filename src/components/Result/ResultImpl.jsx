import React from 'react';
import PropTypes from 'prop-types';

import { useLogoApi } from '@eci/utils/useLogoApi';

import defaultImage from '../assets/images/default-image.png';
import Result from './Result';

const ResultImpl = ({ title, href, status, pubRegNum, logo }) => {
  if (logo && logo.id && logo.mimeType) {
    const { data } = useLogoApi(logo.id);

    return (
      <Result
        title={title}
        href={href}
        status={status}
        pubRegNum={pubRegNum}
        background={`data:${logo.mimeType};base64,${data}`}
      />
    );
  }

  return (
    <Result
      title={title}
      href={href}
      status={status}
      pubRegNum={pubRegNum}
      background={defaultImage}
    />
  );
};

ResultImpl.propTypes = {
  title: PropTypes.string,
  href: PropTypes.string,
  status: PropTypes.string,
  pubRegNum: PropTypes.string,
  logo: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    mimeType: PropTypes.string,
    size: PropTypes.number,
  }),
};

export default ResultImpl;
