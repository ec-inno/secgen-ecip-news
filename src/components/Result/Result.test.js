import React from 'react';
import renderer from 'react-test-renderer';
import Result from './Result';

describe('Result', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Result />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
