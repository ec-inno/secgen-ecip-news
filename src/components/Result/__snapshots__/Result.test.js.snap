// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result renders correctly 1`] = `
<article
  className="ecl-u-d-flex ecl-u-border-bottom ecl-u-border-color-grey-15 ecl-u-pv-m"
>
  <div
    className="ecl-u-flex-grow-1"
  >
    <div
      className="ecl-u-type-prolonged-m ecl-u-type-bold ecl-u-mt-xs"
    >
      <a
        className="ecl-u-type-prolonged-m ecl-u-type-bold ecl-link"
      />
    </div>
    <ul
      className="ecl-u-mt-m ecl-unordered-list ecl-unordered-list--no-bullet"
    >
      <li
        className="ecl-u-d-flex ecl-u-align-items-center ecl-u-type-color-grey-75 ecl-unordered-list__item"
      >
        <svg
          aria-hidden="true"
          className="ecl-icon--m ecl-icon ecl-icon--m"
          focusable="false"
        >
          
          
          <use
            xlinkHref="file#general--organigram"
          />
        </svg>
        <span
          className="ecl-u-type-s ecl-u-ml-s"
        >
          Current status
          : 
        </span>
      </li>
      <li
        className="ecl-u-d-flex ecl-u-align-items-center ecl-u-type-color-grey-75 ecl-unordered-list__item"
      >
        <svg
          aria-hidden="true"
          className="ecl-icon--m ecl-icon ecl-icon--m"
          focusable="false"
        >
          
          
          <use
            xlinkHref="file#general--edit"
          />
        </svg>
        <span
          className="ecl-u-type-s ecl-u-ml-s"
        >
          Registration number
          : 
        </span>
      </li>
    </ul>
  </div>
  <div
    className="ecl-u-media-ratio-3-2 ecl-u-flex-shrink-0 ecl-u-ml-xl ecl-u-d-lg-none"
    role="img"
    style={
      Object {
        "backgroundImage": "url('undefined')",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "contain",
        "height": "5rem",
        "width": "7.5rem",
      }
    }
  />
  <div
    className="ecl-u-media-ratio-3-2 ecl-u-flex-shrink-0 ecl-u-ml-xl ecl-u-d-none ecl-u-d-lg-block"
    role="img"
    style={
      Object {
        "backgroundImage": "url('undefined')",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "contain",
        "height": "8.75rem",
        "width": "13.125rem",
      }
    }
  />
</article>
`;
